data/redisrepo/order_iter.go

```diff
@@ -0,0 +1,82 @@
+package redisrepo
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/orbs-network/order-book/models"
+	"github.com/orbs-network/order-book/utils/logger"
+	"github.com/orbs-network/order-book/utils/logger/logctx"
+)
+
+// ////////////////////////////////////////////////
+type OrderIter struct {
+	index int
+	ids   []string
+	redis *redisRepository
+}
+
+func (i *OrderIter) Next() *models.Order {
+	ctx := context.Background()
+
+	if i.index >= len(i.ids) {
+		logctx.Error(ctx, "Error iterator reached last element")
```

Comment:
I would say this should be info or maybe warning as it is not unexpected behaviour?